cli/internal/lockfile/npm_lockfile.go

```diff
@@ -0,0 +1,232 @@
+	DevDependencies      map[string]string `json:"devDependencies,omitempty"`
+	PeerDependencies     map[string]string `json:"peerDependencies,omitempty"`
+	OptionalDependencies map[string]string `json:"optionalDependencies,omitempty"`
+
+	Bin     map[string]string `json:"bin,omitempty"`
+	Engines map[string]string `json:"engines,omitempty"`
+	CPU     []string          `json:"cpu,omitempty"`
+	OS      []string          `json:"os,omitempty"`
+
+	// Only used for root level package
+	Workspaces []string `json:"workspaces,omitempty"`
+}
+
+// NpmDependency Legacy representation of dependencies
+type NpmDependency struct {
+	Version      string                   `json:"version"`
+	Resolved     string                   `json:"resolved,omitempty"`
+	Integrity    string                   `json:"integrity,omitempty"`
+	Bundled      bool                     `json:"bundled,omitempty"`
+	Dev          bool                     `json:"dev,omitempty"`
+	Optional     bool                     `json:"optional,omitempty"`
+	Requires     map[string]string        `json:"requires,omitempty"`
+	Dependencies map[string]NpmDependency `json:"dependencies,omitempty"`
+}
+
+var _ Lockfile = (*NpmLockfile)(nil)
+
+// ResolvePackage Given a workspace, a package it imports and version returns the key, resolved version, and if it was found
+func (l *NpmLockfile) ResolvePackage(workspacePath turbopath.AnchoredUnixPath, name string, version string) (Package, error) {
+	_, ok := l.Packages[workspacePath.ToString()]
+	if !ok {
+		return Package{}, fmt.Errorf("No package found in lockfile for '%s'", workspacePath)
+	}
+
+	// AllDependencies will return a key to avoid choosing the incorrect transitive dep
+	if entry, ok := l.Packages[name]; ok {
+		return Package{
+			Key:     name,
+			Version: entry.Version,
+			Found:   true,
+		}, nil
+	}
+
+	// If we didn't find the entry just using name, then this is an initial call to ResolvePackage
+	// based on information coming from internal packages' package.json
+	// First we check if the workspace uses a nested version of the package
+	nestedPath := fmt.Sprintf("%s/node_modules/%s", workspacePath, name)
+	if entry, ok := l.Packages[nestedPath]; ok {
+		return Package{
+			Key:     nestedPath,
+			Version: entry.Version,
+			Found:   true,
+		}, nil
+	}
+
+	// Next we check for a top level version of the package
+	hoistedPath := fmt.Sprintf("node_modules/%s", name)
+	if entry, ok := l.Packages[hoistedPath]; ok {
+		return Package{
+			Key:     hoistedPath,
+			Version: entry.Version,
+			Found:   true,
+		}, nil
+	}
+
+	return Package{}, nil
+}
+
+// AllDependencies Given a lockfile key return all (dev/optional/peer) dependencies of that package
+func (l *NpmLockfile) AllDependencies(key string) (map[string]string, bool) {
+	entry, ok := l.Packages[key]
+	if !ok {
+		return nil, false
+	}
+	deps := make(map[string]string, len(entry.Dependencies)+len(entry.DevDependencies)+len(entry.PeerDependencies)+len(entry.OptionalDependencies))
```

Comment:
`go fmt` doesn't like spaces between `+` operators?

Reply:
Apparently not?